taichi/ir/ir.h

```diff
@@ -754,6 +755,16 @@ template <typename T>
 inline void StmtFieldManager::operator()(const char *key, T &&value) {
   if constexpr (std::is_same<typename std::decay<T>::type, Stmt *>::value) {
     stmt->add_operand(const_cast<Stmt *&>(value));
+  } else if constexpr (std::is_same<typename std::decay<T>::type,
+      std::vector<Stmt *> >::value) {
+    auto operand_stmts = const_cast<std::vector<Stmt *> &>(value);
```

Comment:
Please try using `auto &operand_stmts` instead of `auto operand_stmts` here.

Reply:
(Note that `auto a = b` always creates a copy of `b`. Then you are returning pointers to temporary local variables, which is undefined behavior after the function returns...)

Reply:
Oh I thought `auto a = const_cast<A&> b` as the same as `A &a = const_cast<A&> b`. If I thought more carefully, C++ cannot deduce whether it's an lvalue reference type unless I specified `auto &`, since `const_cast<A&> b` is merely an lvalue. Thanks!

Reply:
Exactly!